homeassistant/components/lektrico/binary_sensor.py

```diff
@@ -0,0 +1,118 @@
+"""Support for Lektrico charging station binary sensors."""
+
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_IDENTIFIERS,
+    ATTR_MANUFACTURER,
+    ATTR_MODEL,
+    ATTR_NAME,
+    ATTR_SW_VERSION,
+    CONF_FRIENDLY_NAME,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import LektricoDeviceDataUpdateCoordinator
+from .const import DOMAIN
+
+
+@dataclass
+class LektricoBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """A class that describes the Lektrico binary sensor entities."""
+
+    @classmethod
+    def get_is_on(cls, data: Any) -> bool | None:
+        """Return None."""
+        return None
+
+
+@dataclass
+class HeadlessBinarySensorEntityDescription(LektricoBinarySensorEntityDescription):
+    """A class that describes the Lektrico No Authentication binary sensor entity."""
+
+    @classmethod
+    def get_is_on(cls, data: Any) -> bool:
+        """Get the headless."""
+        return bool(data.headless)
+
+
+SENSORS: tuple[LektricoBinarySensorEntityDescription, ...] = (
+    HeadlessBinarySensorEntityDescription(
+        key="headless",
+        name="No Authentication",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Lektrico charger based on a config entry."""
+    _lektrico_device: LektricoDeviceDataUpdateCoordinator = hass.data[DOMAIN][
+        entry.entry_id
+    ]
+    sensors = [
+        LektricoBinarySensor(
+            sensor_desc,
+            _lektrico_device,
+            entry.data[CONF_FRIENDLY_NAME],
+        )
+        for sensor_desc in SENSORS
+    ]
+
+    async_add_entities(sensors, False)
+
+
+class LektricoBinarySensor(CoordinatorEntity, BinarySensorEntity):
+    """The entity class for Lektrico charging stations binary sensors."""
+
+    entity_description: LektricoBinarySensorEntityDescription
```

Comment:
To indicate this integration use modern entity naming

```suggestion
    entity_description: LektricoBinarySensorEntityDescription
    _attr_has_entity_name = True
```

Reply:
I made the changes to use modern entity naming, thanks.